device/device.go

```diff
@@ -43,3 +60,109 @@ func Init(typeFlag string) {
 		return
 	}
 }
+
+// New triggers select repository methods and initializes a new flasher
+func New(device string) (Flasher, error) {
+	g := make([]string, 0)
+
+	if device == CustomFlash {
+		r, err := repo.GetAllRepos()
+		if err != nil {
+			return nil, err
+		}
+
+		for _, s := range r {
+			c := true
+			for _, d := range devices {
+				if s == d {
+					c = false
+				}
+			}
+			if c {
+				g = append(g, s)
+			}
+		}
+
+		if len(g) != 0 {
+			device = g[dialogs.SelectOneDialog("Please select your custom board: ", g)]
+		}
+
+	}
+	r := &repo.DeviceMapping{}
```

Comment:
__[gometalinter]__
* [ineffassign] ineffectual assignment to r

<!-- Comment by [SideCI](https://sideci.com) -->